packages/sandbox/src/lambda_function_log_streamer.ts

```diff
@@ -0,0 +1,185 @@
+import { LogLevel, Printer } from '@aws-amplify/cli-core';
+import { BackendOutputClient } from '@aws-amplify/deployed-backend-client';
+import {
+  BackendIdentifierConversions,
+  TagName,
+} from '@aws-amplify/platform-core';
+import { BackendIdentifier, BackendOutput } from '@aws-amplify/plugin-types';
+import {
+  CloudFormationClient,
+  DescribeStacksCommand,
+} from '@aws-sdk/client-cloudformation';
+import { LambdaClient, ListTagsCommand } from '@aws-sdk/client-lambda';
+import { CloudWatchLogEventMonitor } from './cloudwatch_logs_monitor.js';
+import { build as buildArn, parse as parseArn } from '@aws-sdk/util-arn-parser';
+import { SandboxFunctionStreamingOptions } from './sandbox.js';
+
+/**
+ * Logs streamer for customer defined lambda functions in a sandbox.
+ */
+export class LambdaFunctionLogStreamer {
+  private enabled: boolean = false;
+  /**
+   * Creates an instance of LambdaFunctionLogStreamer
+   */
+  constructor(
+    private readonly lambda: LambdaClient,
+    private readonly cfnClient: CloudFormationClient,
+    private readonly logsMonitor: CloudWatchLogEventMonitor,
+    private readonly backendOutputClient: BackendOutputClient,
+    private readonly printer: Printer
+  ) {}
+
+  /**
+   * Starts streaming logs in the given sandbox.
+   * @param sandboxBackendId The sandbox backend identifier.
+   * @param streamingOptions Options to configure the log streaming.
+   */
+  startStreamingLogs = async (
+    sandboxBackendId: BackendIdentifier,
+    streamingOptions?: SandboxFunctionStreamingOptions
```

Comment:
Since these parameters don't change over the lifecycle of this class, it seems like they should be part of the ctor